src/prebid.js

```diff
@@ -384,43 +351,30 @@ $$PREBID_GLOBAL$$.requestBids = function ({ bidsBackHandler, timeout, adUnits, a
     adUnit.bids = adUnit.bids.filter(nativeBidder);
   });
 
-  if (auctionRunning) {
-    bidRequestQueue.push(() => {
-      $$PREBID_GLOBAL$$.requestBids({ bidsBackHandler, timeout: cbTimeout, adUnits, adUnitCodes });
-    });
-    return;
-  }
-
-  auctionRunning = true;
-
-  // we will use adUnitCodes for filtering the current auction
-  $$PREBID_GLOBAL$$._adUnitCodes = adUnitCodes;
-
-  bidmanager.externalCallbackReset();
-  clearPlacements();
-
   if (!adUnits || adUnits.length === 0) {
     utils.logMessage('No adUnits configured. No bids requested.');
     if (typeof bidsBackHandler === 'function') {
-      bidmanager.addOneTimeCallback(bidsBackHandler, false);
+      // executeCallback, this will only be called in case of first request
+      try {
+        bidsBackHandler();
+      } catch (e) {
+        utils.logError('Error executing bidsBackHandler', null, e);
+      } finally {
+        $$PREBID_GLOBAL$$.clearAuction();
+      }
     }
-    bidmanager.executeCallback();
     return;
   }
 
-  // set timeout for all bids
-  const timedOut = true;
-  const timeoutCallback = bidmanager.executeCallback.bind(bidmanager, timedOut);
-  const timer = setTimeout(timeoutCallback, cbTimeout);
-  setAjaxTimeout(cbTimeout);
-  if (typeof bidsBackHandler === 'function') {
-    bidmanager.addOneTimeCallback(bidsBackHandler, timer);
-  }
+  const auction = auctionManager.createAuction();
+  auction.setAdUnits(adUnits);
+  auction.setAdUnitCodes(adUnitCodes);
 
-  adaptermanager.callBids({ adUnits, adUnitCodes, cbTimeout });
-  if ($$PREBID_GLOBAL$$._bidsRequested.length === 0) {
-    bidmanager.executeCallback();
+  if (typeof bidsBackHandler === 'function') {
+    auction.startAuctionTimer(bidsBackHandler, cbTimeout);
```

Comment:
I may be missing something... but this looks buggy to me.

The bidsBackHandler is supposed to be called whenever all the bids are back... and I thought that the expected argument was the `bids`.

...but here, you're passing it in as an argument where the AuctionManager calls `setCallback`. Then it looks like`callBids` is sending in `this.done()` as the final "done" callback to the `adapterManager`. `this.done` eventually calls `this.executeCallback()` once it receives all the bids. `executeCallback` then calls `getCallback()`, and then `callback($$PREBID_GLOBAL$$)`.

So... it looks to me like the `bidsBackHandler` may be called with $$PREBID_GLOBAL$$ as an argument, rather than the bids themselves.